storage/src/nodestore/mod.rs

```diff
@@ -472,6 +472,12 @@ pub struct NodeStore<T, S> {
     storage: Arc<S>,
 }
 
+impl<T, S> NodeStore<T, S> {
+    pub(crate) fn storage(&self) -> &S {
+        self.storage.as_ref()
+    }
+}
+
```

Comment:
nit: the surface area seems to be getting bigger here. Can this be better abstracted into something like `fn read_freed_area` on NodeStore instead of doing it here? Then it could live in `alloc.rs`.